firedrake/pointquery_utils.py

```diff
@@ -122,8 +122,28 @@ def init_X(fiat_cell, parameters):
     vertices = numpy.array(fiat_cell.get_vertices())
     X = numpy.average(vertices, axis=0)
 
-    formatter = ArrayInit(X, precision=numpy.finfo(parameters["scalar_type"]).resolution)._formatter
-    return "\n".join("%s = %s;" % ("X[%d]" % i, formatter(v)) for i, v in enumerate(X))
+    precision = numpy.finfo(parameters["scalar_type"]).resolution
+    return "\n".join("%s = %s;" % ("X[%d]" % i, format_num(v, precision)) for i, v in enumerate(X))
+
+
+def format_num(v, precision):
+    """Format a real or complex value into a string, showing up to
+    ``precision`` decimal digits.  This function is partly
+    extracted from the open_source "FFC: the FEniCS Form
+    Compiler", freely accessible at
+    https://bitbucket.org/fenics-project/ffc.
+    """
+    f = "%%.%dg" % precision
```

Comment:
Yuck

Reply:
I think that this is more bizarre rounding behaviour. I will delete it and see if the tests still pass.

Reply:
They do!